app/controllers/users_controller.rb

```diff
@@ -0,0 +1,63 @@
+
+class UsersController < ApplicationController
+
+use Rack::Flash
+
+
+
+get '/signup' do 
+  if logged_in?
+    redirect '/tweets'
+  else
+  erb :'users/create_user'
+  end
+end
+
+post "/signup" do 
+  user = User.new(params)
+    if user.save
+      session[:user_id] = user.id
+      redirect to '/tweets'
+    else
+      flash[:message] = "Please enter an email, username and password to create an account. "
+      redirect to "/signup"
+    end
+end
+
+
+get '/login' do
+  if logged_in?
+    redirect '/tweets'
+  else
+  erb :'users/login'
+  end
+end
+
+post "/login" do
+  #raise params.inspect
+  user = User.find_by(username: params[:username])
```

Comment:
That's a great point. I should validate the uniqueness of the username in the user model. So I just added:
validates_uniqueness_of :username
thanks.